### PR TITLE
feat: support multi-period live DASH with changing periods

### PR DESCRIPTION
**Pull request by gesinger**
In order to handle live DASH manifests where periods are changed, added, and removed,
merging is used to ensure that the new manifest is a continuation of the last parsed
manifest.

In order to handle live cases, the last MPD must be provided via the `lastMpd` option.

BREAKING CHANGE: period start time is now used as the timeline instead of period index

### CODE REVIEW

File: src/utils/list.js [hunk 1]
```diff
@@ -31,3 +31,55 @@ export const findIndexes = (l, key) => l.reduce((a, e, i) => {
...
+/**
+ * Returns a union of the included lists provided each element can be identified by a key.
+ *
+ * @param {Array} list - list of lists to get the union of
+ * @param {Function} keyFunction - the function to use as a key for each element
+ *
+ * @return {Array} the union of the arrays
+ */
+export const union = (lists, keyFunction) => {
+  return Object.values(lists.reduce((acc, list) => {
```
Comment (gkatsev) [inline src/utils/list.js:78]: `Object.values` doesn't exist on IE11 either. Probably should use `Object.keys()` and loop through that.

**Review by gkatsev — COMMENTED** [review]
Quick comments on the small files.

File: src/toM3u8.js [hunk 2]
```diff
@@ -30,14 +36,22 @@ const mergeDiscontiguousPlaylists = playlists => {
     } else {
       // first Period
       acc[name] = playlist;
+      acc[name].attributes.timelineStarts = [];
```
Comment (brandonocasey) [inline src/toM3u8.js:39]: seems more logical for this code to be the first if statement and the else to be for "Periods after first"

**Review by gkatsev — COMMENTED** [review]
Some fairly minor comments.

**Review by gkatsev — APPROVED** [review]

**Review by gkatsev — APPROVED** [review]
Code LGTM, haven't tested it yet.

**Review by gkatsev — APPROVED** [review]
Just ran a bunch of tests and it works great!

**Pull request merged**